pkg/operator/controllers/workaround/systemreserved.go

```diff
@@ -0,0 +1,197 @@
+package workaround
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"encoding/json"
+	"fmt"
+
+	azureproviderv1beta1 "github.com/openshift/cluster-api-provider-azure/pkg/apis/azureprovider/v1beta1"
+	clusterapi "github.com/openshift/cluster-api/pkg/client/clientset_generated/clientset"
+	mcv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	mcoclient "github.com/openshift/machine-config-operator/pkg/generated/clientset/versioned"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/util/retry"
+
+	"github.com/Azure/ARO-RP/pkg/dynamichelper"
+	"github.com/Azure/ARO-RP/pkg/util/version"
+)
+
+type systemreserved struct {
+	mcocli     mcoclient.Interface
+	clustercli clusterapi.Interface
+	dh         dynamichelper.DynamicHelper
+	log        *logrus.Entry
+}
+
+const (
+	labelName                  = "bz-1857446"
+	labelValue                 = "hotfixed"
+	machineConfigName          = "bz-1857446-hotfix"
+	workerMachineSetsNamespace = "openshift-machine-api"
+)
+
+var (
+	_          Workaround = &systemreserved{}
+	vmCapacity            = map[string]map[string]string{
+		"Standard_D16as_v4": {"MemGB": "64", "vCPUs": "16"},
+		"Standard_D16s_v3":  {"MemGB": "64", "vCPUs": "16"},
+		"Standard_D2s_v3":   {"MemGB": "8", "vCPUs": "2"},
+		"Standard_D32as_v4": {"MemGB": "128", "vCPUs": "32"},
+		"Standard_D32s_v3":  {"MemGB": "128", "vCPUs": "32"},
+		"Standard_D4as_v4":  {"MemGB": "16", "vCPUs": "4"},
+		"Standard_D4s_v3":   {"MemGB": "16", "vCPUs": "4"},
+		"Standard_D8as_v4":  {"MemGB": "32", "vCPUs": "8"},
+		"Standard_D8s_v3":   {"MemGB": "32", "vCPUs": "8"},
+		"Standard_E16s_v3":  {"MemGB": "128", "vCPUs": "16"},
+		"Standard_E32s_v3":  {"MemGB": "256", "vCPUs": "32"},
+		"Standard_E4s_v3":   {"MemGB": "32", "vCPUs": "4"},
+		"Standard_E8s_v3":   {"MemGB": "64", "vCPUs": "8"},
+		"Standard_F16s_v2":  {"MemGB": "32", "vCPUs": "16"},
+		"Standard_F32s_v2":  {"MemGB": "64", "vCPUs": "32"},
+		"Standard_F4s_v2":   {"MemGB": "8", "vCPUs": "4"},
+		"Standard_F8s_v2":   {"MemGB": "16", "vCPUs": "8"},
+	}
+	// convert total memory on the VM to amount to reserve
+	memReserved = map[string]string{
+		"8":   "1.8",
+		"16":  "2.6",
+		"32":  "3.56",
+		"64":  "5.48",
+		"128": "9.32",
+		"256": "11.88",
+	}
+	// convert vCPUs to millicores to reserve
+	cpuReserved = map[string]string{
+		"2":  "100",
+		"4":  "140",
+		"8":  "180",
+		"16": "260",
+		"32": "420",
+		"64": "740",
```

Comment:
Per @mjudeikis's update on wiki, OpenShift default is already 500m, so this should be 500m unless 16+ cores where we should use 750m.